microceph/cmd/microceph/disk_add.go

```diff
@@ -64,28 +85,116 @@ func (c *cmdDiskAdd) Run(cmd *cobra.Command, args []string) error {
+	if c.flagAllDevices {
+		disks, err := getUnpartitionedDisks(cli)
+		if err != nil {
+			return err
+		}
+
+		// Prepare Batch arguments
+		for _, disk := range disks {
+			req.Path = append(req.Path, disk.Path)
+		}
+	} else {
+		// Pass space separated params as disk paths.
+		req.Path = args
+
+		if !strings.Contains(req.Path[0], common.LoopSpecId) {
+			if c.walDevice != "" {
+				req.WALDev = &c.walDevice
+				req.WALWipe = c.walWipe
+				req.WALEncrypt = c.walEncrypt
+			}
+
+			if c.dbDevice != "" {
+				req.DBDev = &c.dbDevice
+				req.DBWipe = c.dbWipe
+				req.DBEncrypt = c.dbEncrypt
+			}
+		}
 	}
 
-	if c.walDevice != "" {
-		req.WALDev = &c.walDevice
-		req.WALWipe = c.walWipe
-		req.WALEncrypt = c.walEncrypt
+	// required request params.
+	req.Wipe = c.flagWipe
+	req.Encrypt = c.flagEncrypt
+	failures, err := client.AddDisk(context.Background(), cli, &req)
+	if err != nil {
+		return err
 	}
 
-	if c.dbDevice != "" {
-		req.DBDev = &c.dbDevice
-		req.DBWipe = c.dbWipe
-		req.DBEncrypt = c.dbEncrypt
+	err = printAddDiskFailures(failures)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func printAddDiskFailures(response types.DiskAddResponse) error {
+	var errStr string
+	data := [][]string{}
+
+	if len(response.ValidationError) != 0 {
+		fmt.Println("Validation Error found")
+		return fmt.Errorf(response.ValidationError)
 	}
 
-	err = client.AddDisk(context.Background(), cli, req)
+	if len(response.Reports) == 0 {
+		// No responses; nothing to do.
+		return nil
+	}
+
+	failureCount := 0
+	for _, report := range response.Reports {
+		if strings.Contains(report.Report, "Failure") {
+			failureCount += 1
+			errStr = report.Error
+		}
+		// prepare data for tabulated display.
+		data = append(data, []string{report.Path, report.Report})
+	}
+
+	// Print disk add failures in tabulated form.
+	fmt.Println("")
+	header := []string{"Path", "Status"}
+	sort.Sort(lxdCmd.SortColumnsNaturally(data))
+	err := lxdCmd.RenderTable(lxdCmd.TableFormatTable, header, data, data)
 	if err != nil {
 		return err
 	}
 
+	if failureCount == 1 {
+		// Print error if only one instance of error is there.
+		fmt.Println("Error: ", errStr)
+	} else if failureCount > 1 {
+		// Print error if only one instance of error is there.
+		fmt.Println("Failed adding multiple disks, please check logs for details.")
+	}
+
+	return nil
+}
+
+// validateBatchArgs checks if no loop spec is provided as an argument to batch disk addition.
+func (c *cmdDiskAdd) validateBatchArgs(args []string) error {
+	// no validation if single arg is provided.
+	if len(args) == 1 {
+		return nil
+	}
+
+	// if wal/db devices are provided with batch commands.
+	if c.walDevice != "" {
+		return fmt.Errorf("--wal-device flag is not supported for batch disk addition")
+	}
+
+	if c.dbDevice != "" {
+		return fmt.Errorf("--db-device flag is not supported for batch disk addition")
+	}
+
+	for _, diskPath := range args {
+		if strings.Contains(diskPath, common.LoopSpecId) {
```

Comment:
Why not `strings.HasPrefix(diskPath, common.LoopSpecId)`?

Reply:
That's a good nit. I've changed all 4 occurrences of this check to only look for prefixes.